From cfb889ab38befe1e378d92ab9915aed14ec5843b Mon Sep 17 00:00:00 2001
From: "Friedrich W. H. Kossebau" <kossebau@kde.org>
Date: Wed, 8 Dec 2021 20:29:29 +0100
Subject: [PATCH] Use KDECompilerSettings instead of
 KDEFrameworkCompilerSettings

The latter is only for KF modules to use.

Also include the KDE cmake modules as first thing, as their settings
can influence other included macros.

GIT_SILENT
---
 CMakeLists.txt | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b89f16..42f1b5b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,18 +14,25 @@ set(QT_MIN_VERSION "5.15.0")
 set(KF5_MIN_VERSION "5.85.0")
 
 
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core PrintSupport TextToSpeech Xml Widgets)
-
 find_package(ECM ${KF5_MIN_VERSION} NO_MODULE REQUIRED)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH})
 
-include(ECMAddAppIcon)
 include(KDEInstallDirs)
 include(KDECMakeSettings)
-include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
+include(KDECompilerSettings NO_POLICY_SCOPE)
+
+include(ECMAddAppIcon)
 include(ECMInstallIcons)
 include(ECMAddAppIcon)
 
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS
+    Core
+    PrintSupport
+    TextToSpeech
+    Xml
+    Widgets
+)
+
 find_package(KF5  ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     Crash
     Completion
-- 
GitLab

